src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -91,6 +91,44 @@ def zero(cls) -> Self:
         assert isinstance(_zero, cls)
         return _zero
 
+    @classmethod
+    def from_string(cls, string: str) -> Self:
+        """Return a quantity from a string representation.
+
+        Args:
+            string: The string representation of the quantity.
+
+        Returns:
+            A quantity object with the value given in the string.
+
+        Raises:
+            ValueError: If the string does not match the expected format.
+
+        """
+        split_string = string.split(" ")
+
+        if len(split_string) != 2:
+            raise ValueError(
+                f"Expected a string of the form 'value unit', got {string}"
+            )
+
+        assert cls._exponent_unit_map is not None
+        exp_map = cls._exponent_unit_map
+
+        for exponent, unit in exp_map.items():
+            if unit == split_string[1]:
+                instance = cls.__new__(cls)
+                try:
+                    instance._base_value = float(split_string[0]) * 10**exponent
+                except ValueError as error:
+                    raise ValueError(
+                        f"Failed to parse string {string}: {error}"
+                    ) from error
```

Comment:
Not a biggie, but if you are raising the exception `from` another `error`, is redundant to include `error` in the message string, as if you need to use it programmatically you can access the chained errors and if the exception is unhandled, Python will show the whole exception chain too.